rules/rulelist.go

```diff
@@ -81,6 +81,7 @@ func Generate(filters ...RuleFilter) RuleList {
 		{"G303", "Creating tempfile using a predictable path", NewBadTempFile},
 		{"G304", "File path provided as taint input", NewReadFile},
 		{"G305", "File path traversal when extracting zip archive", NewArchive},
+		{"G307", "Unsafe defer of closing a file", NewDeferredClosing},
```

Comment:
I would change this message to something like: `Unsafe defer call of a method returning an error`

Reply:
Changes made.